### PR TITLE
[RELAY][RUNTIME] Add compute and schedule attributes for all ops in relay/op/tensor.py

### PR DESCRIPTION
**Pull request by jroesch**
This PR ports over schedule and compute attributes for the operators in `tensor.py`. I hope this will serve as an example of adding this behavior for operators so others in the community can help continue the work of making Relay operator complete. 

I simply copied over the same scheduling primitives from NNVM, and am using out of the box compute functions. Feedback much appreciated.

For details on helping with the rest of the operators please read here #2051.

cc @tqchen @MarisaKirisame @slyubomirsky @zhiics @siju-samuel 

There are two missing cases (concatenate and copy both of which require a little more work).

### CODE REVIEW

**Comment by MarisaKirisame** [comment]
this code seems way too highly repetitive to me.
this is a typical snippet of _tensor.py, with every other line being virtually like this.
`def sqrt_compute(attrs, inputs, output_type, target):
    assert len(inputs) == 1
    return [topi.sqrt(inputs[0])]
 register_compute("sqrt", sqrt_compute)
register_schedule("sqrt", schedule_broadcast)`
there are 3 things:
define, which assert len and use topi
register_compute
register_schedule

register_compute, register_schedule could be merged into one single function
sqrt_compute can be made into one line by writing a hof that take two argument (the topi func and num of input)

If I were to write this, I will declare a table that store the name, and the num of input, traverse it, and define function with metaprogramming, to minimize the chance of error, and to maximize reuse(other code that build ontop of relay can simply read the table).
Of course, it might be going too far for tvm coding style, but fusing register_compute/register_schedule is certainly doable, and so is using a hof.

**Comment by jroesch** [comment]
@MarisaKirisame the argument against that style (which is roughly) what NNVM has done is that it becomes much harder for users to read and modify. Most code is effectively read-only, and especially code like this will not change often. I think it is better to err on repetitive and readable than inscrutable.

File: python/tvm/relay/op/_tensor.py [hunk 1]
```diff
@@ -1,49 +1,294 @@
...
+    assert len(inputs) == 2
+    return [topi.full(inputs[0], inputs[1], 0.0)]
+
+register_compute("zeros_compute", zeros_compute)
+register_schedule("zeros_compute", schedule_injective)
+
+# zeros_like
+def zeros_like_compute(attrs, inputs, output_type, target):
+    assert len(inputs) == 1
+    return [topi.full_like(inputs[0], 0.0)]
```
Comment (tqchen) [inline python/tvm/relay/op/_tensor.py]: the output data type can likely be wrong here

File: python/tvm/relay/op/_tensor.py [hunk 2]
```diff
@@ -1,49 +1,294 @@
 #pylint: disable=invalid-name, unused-argument
 """Backend compiler related feature registration"""
+from __future__ import absolute_import
 import tvm
 import topi
+import topi.cuda
 from . import register
 
+def register_schedule(op_name, schedule):
```
[the remark this answers is not in the document]
Comment (tqchen) [reply python/tvm/relay/op/_tensor.py]: All the registry functions should goto op.registry.py

File: python/tvm/relay/op/_tensor.py [hunk 3]
```diff
@@ -1,49 +1,294 @@
...
+
+register_compute("less", less_compute)
+register_schedule("less", schedule_broadcast)
+
+# less equal
+def less_equal_compute(attrs, inputs, output_type, target):
+    assert len(inputs) == 2
+    return [topi.less_equal(inputs[0], inputs[1])]
+
+register_compute("less_equal", less_equal_compute)
```
Comment (tqchen) [inline python/tvm/relay/op/_tensor.py]: As per nnvm tradition, we should move most of the compute function to c++, and only leave a few(e.g. conv2d) in python

Comment (jroesch) [reply python/tvm/relay/op/_tensor.py]: Our current plan is to do this in a second pass.

File: include/tvm/relay/module.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+#include <tvm/relay/type.h>
+#include <string>
+#include <vector>
+
+namespace tvm {
+namespace relay {
+
+struct Module;
+
+/*! \brief The global environment of Relay programs.
```
Comment (zhiics) [inline include/tvm/relay/module.h]: s/environment/module?

File: include/tvm/relay/module.h [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+   * \returns The function named by the argument.
+   */
+  Function Lookup(const std::string& name);
+
+  /*!
+   * \brief Update the functions inside this environment by
+   *        functions in another environment.
+   * \param other The other environment.
+   */
+  void Update(const Module& other);
```
Comment (zhiics) [inline include/tvm/relay/module.h]: Move `Update` up before `Remove` ?

File: include/tvm/relay/module.h [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+   * ensures global uniqueness.
+   */
+  tvm::Map<std::string, GlobalVar> global_var_map_;
+};
+
+struct Module : public NodeRef {
+  Module() {}
+  explicit Module(NodePtr<tvm::Node> p) : NodeRef(p) {}
+
+  inline ModuleNode* operator->() const {
```
Comment (zhiics) [inline include/tvm/relay/module.h]: no need `inline` for members defined in class.

**Comment by zhiics** [comment]
It seems that we have several classes named as `Module` including the ones from runtime. I am totally okay with them, but would it be a little confusing for TVM beginners? But again, I like the renaming from `Environment` to `Module` as it is more friendly to those who are familiar with LLVM.

**Comment by tqchen** [comment]
Merge this in and will update the follow-up on #2059 to depend on this

**Pull request merged**